services/clients/github_client.py

```diff
@@ -27,7 +28,8 @@ def read_file(self, github_url: str) -> str:
 
         data = request.json()
         if "type" in data and data["type"] == "dir":
-            raise ValueError(f"{github_url} is a folder, not a file.")
+            raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST,
```

Comment:
Will this need to be updated for the repo apis?

Reply:
I imagine we will restructure this or maybe create a different function for folders. We will need something to control that whole flow, since folders depend on the documentation of files under it and all that stuff